### PR TITLE
docs: note required RNTL version for user-event in user-event.md

### PR DESCRIPTION
**Pull request by meatnordrink**
### Summary
I had to go through the changelog to figure out what version of RNTL I needed to use user-event, as simply being on the latest major isn't sufficient (12.0.1). I thought this might save others time/frustration.

### Test plan
Make sure the updated docs look okay and are clear.

### CODE REVIEW

**Review by pierrezimmermannbam — APPROVED** [review]
@meatnordrink thank you very much for the contribution, it's a great idea! You'll need to wait for another maintainer's approval before this can be merged though

**Review by mdjastrzebski — APPROVED** [review]
Thanks for PR!

**Pull request merged**
